src/reducers/filter.js

```diff
@@ -0,0 +1,13 @@
+function filter(state = null, action) {
```

Comment:
The default state should be 'ALL', not `null`.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/ItDefaultsToAll.png)

You migth be able to simplify the logic you have here if you default to 'All' instead of null.